documents/sd/chapter/Classes/Common/Events/Events.tex

```diff
@@ -11,101 +11,314 @@ \subsection*{MORR.Shared.Events}
+	
+	\clsdcl{public abstract class DefaultEncodeableEventQueue<T> : BaseEventQueue<T>, IEncodableEventQueue<T> where T : Event}
+	
+	\clsdsp{Provides an event queue for events intended for encoding.}
+\end{absclass}
 
-    \clsdsp{Provides read-only access to a queue of events as concrete type T.}
+\begin{absclass}{DefaultDecodableEventQueue<T>}
+	\clsdiagram[width=0.6\textwidth]{resources/Classes/Common/Shared/Events/Queue/DefaultDecodableEventQueue.png}
+	
+	\clsdcl{public abstract class DefaultDecodableEventQueue<T> : BaseEventQueue<T>, IDecodableEventQueue<T> where T : Event}
+	
+	\clsdsp{Provides an event queue for events intended for decoding.}
+\end{absclass}
+
+\begin{absclass}{NonDeserializableEventQueue<T>}
+	\clsdiagram[width=0.6\textwidth]{resources/Classes/Common/Shared/Events/Queue/NonDeserializableEventQueue.png}
+	
+	\clsdcl{public abstract class NonDeserializableEventQueue<T> : BaseEventQueue<T>, 
+	IReadOnlyEventQueue<T> where T : Event}
+	
+	\clsdsp{Provides an event queue for events that do not support deserialization.}
+\end{absclass}
+	
+\begin{absclass}{BaseEventQueue<TEvent>}
+	\clsdiagram[width=0.5\textwidth]{resources/Classes/Common/Shared/Events/Queue/BaseEventQueue.png}
+	
+	\clsdcl{public abstract class BaseEventQueue<TEvent> where TEvent : Event}
+	
+	\clsdsp{Provides the basic structure of an EventQueue.}
+	
+	\begin{attributes}
+		\attribute{public bool IsClosed}{Describes whether the queue is currently enabled to queue new events or not.}
+	\end{attributes}
 
     \begin{methods}
-        \begin{method}{IAsyncEnumerable<T> GetEvents()}{Asynchronously gets all event as their concrete type T.}
-            \return{IAsyncEnumerable<T>}{A stream of events of type T.}
+        \begin{method}{public IAsyncEnumerable<TEvent> GetEvents()}{Asynchronously gets all events as concrete type TEvent.}
+            \return{IAsyncEnumerable<TEvent>}{A stream of events of type T.}
+        \end{method}
+        \begin{method}{public void Enqueue(TEvent @event)}{Asynchronously enqueues a new event.}
+        	\begin{parameters}
+        		\para{TEvent @event}{The event to enqueue}
+        	\end{parameters}
+        \end{method}
+        \begin{method}{public void Open()}{Opens the EventQueue so new events can be queued.}
+        \end{method}
+        \begin{method}{public void Close()}{Closes the EventQueue so no new event can be queued.}
         \end{method}
     \end{methods}
+\end{absclass}
+
+\begin{interface}{IDecodableEventQueue<out T>}
+	\clsdiagram[width=0.5\textwidth]{resources/Classes/Common/Shared/Events/Queue/IDecodableEventQueue.png}
+
+	\clsdcl{public interface IDecodableEventQueue<out T> where T : Event}
+	
+	\clsdsp{Provides a read-only queue for event types intended for decoding.}
+	
+	\begin{attributes}
+		\attribute{public bool IsClosed{get;}}{Describes whether the queue is currently enabled to queue new events or not.}
```

Comment:
Could you escape the "{"? Without this, the formatting is a bit strange, you should be able to just replace it with "IsClosed \\{ get; \\}"